### PR TITLE
Added missing key vault secrets package pipelines

### PR DESCRIPTION
**Pull request by unaizorrilla**
This PR add missing pipelines for new Azure Key Vault Secrets package added recently

### CODE REVIEW

**Review by lfraile — APPROVED** [review]

File: .github/workflows/healthchecks_azurekeyvault_secrets_cd.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            6.0.x
+            7.0.x
+      - name: Restore
+        run: dotnet restore ./src/HealthChecks.AzureKeyVault/HealthChecks.AzureKeyVault.Secrets.csproj
+      - name: Build
+        run: dotnet build --no-restore ./src/HealthChecks.AzureKeyVault.Secrets/HealthChecks.AzureKeyVault.Secrets.csproj -c $BUILD_CONFIG
+      - name: Pack
+        run: dotnet pack --no-build ./src/HealthChecks.AzureKeyVault.Secrets/HealthChecks.AzureKeyVault.Secrets.csproj -c $BUILD_CONFIG -o ./artifacts
+      - name: Publish
+        run: dotnet nuget push ./artifacts/AspNetCore.HealthChecks.AzureKeyVault.Secrets.*.nupkg -k ${{secrets.NUGET_API_KEY}} -s https://api.nuget.org/v3/index.json --skip-duplicate
```
Comment (adamsitnik) [inline .github/workflows/healthchecks_azurekeyvault_secrets_cd.yml]: the paths miss a dot in the name: HealthChecks.Azure**.**KeyVault.Secrets

```suggestion
        run: dotnet restore ./src/HealthChecks.Azure.KeyVault.Secrets/HealthChecks.Azure.KeyVault.Secrets.csproj
      - name: Build
        run: dotnet build --no-restore ./src/HealthChecks.Azure.KeyVault.Secrets/HealthChecks.Azure.KeyVault.Secrets.csproj -c $BUILD_CONFIG
      - name: Pack
        run: dotnet pack --no-build ./src/HealthChecks.Azure.KeyVault.Secrets/HealthChecks.Azure.KeyVault.Secrets.csproj -c $BUILD_CONFIG -o ./artifacts
      - name: Publish
        run: dotnet nuget push ./artifacts/AspNetCore.HealthChecks.Azure.KeyVault.Secrets.*.nupkg -k ${{secrets.NUGET_API_KEY}} -s https://api.nuget.org/v3/index.json --skip-duplicate
```

File: .github/workflows/healthchecks_azurekeyvault_secrets_cd_preview.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            6.0.x
+            7.0.x
+      - name: Restore
+        run: dotnet restore ./src/HealthChecks.AzureKeyVault.Secrets/HealthChecks.AzureKeyVault.Secrets.csproj
+      - name: Build
+        run: dotnet build --no-restore ./src/HealthChecks.AzureKeyVault.Secrets/HealthChecks.AzureKeyVault.Secrets.csproj -c $BUILD_CONFIG
+      - name: Pack
+        run: dotnet pack --no-build ./src/HealthChecks.AzureKeyVault.Secrets/HealthChecks.AzureKeyVault.Secrets.csproj --version-suffix $VERSION_SUFFIX -c $BUILD_CONFIG -o ./artifacts
+      - name: Publish
+        run: dotnet nuget push ./artifacts/AspNetCore.HealthChecks.AzureKeyVault.Secrets.*.nupkg -k ${{secrets.NUGET_API_KEY}} -s https://api.nuget.org/v3/index.json --skip-duplicate
```
Comment (adamsitnik) [inline .github/workflows/healthchecks_azurekeyvault_secrets_cd_preview.yml:30]:
```suggestion
        run: dotnet restore ./src/HealthChecks.Azure.KeyVault.Secrets/HealthChecks.Azure.KeyVault.Secrets.csproj
      - name: Build
        run: dotnet build --no-restore ./src/HealthChecks.Azure.KeyVault.Secrets/HealthChecks.Azure.KeyVault.Secrets.csproj -c $BUILD_CONFIG
      - name: Pack
        run: dotnet pack --no-build ./src/HealthChecks.Azure.KeyVault.Secrets/HealthChecks.Azure.KeyVault.Secrets.csproj --version-suffix $VERSION_SUFFIX -c $BUILD_CONFIG -o ./artifacts
      - name: Publish
        run: dotnet nuget push ./artifacts/AspNetCore.HealthChecks.Azure.KeyVault.Secrets.*.nupkg -k ${{secrets.NUGET_API_KEY}} -s https://api.nuget.org/v3/index.json --skip-duplicate
```

File: .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
+name: HealthChecks AzureKeyVault Secrets CI
+
+on:
+  workflow_dispatch:
+  push:
+   branches: [ master ]
+   paths:
+     - src/HealthChecks.AzureKeyVault.Secrets/**
+     - test/HealthChecks.AzureKeyVault.Secrets.Tests/**
```
Comment (adamsitnik) [inline .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml:9]:
```suggestion
     - src/HealthChecks.Azure.KeyVault.Secrets/**
     - test/HealthChecks.Azure.KeyVault.Secrets.Tests/**
```

File: .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+     - Directory.Build.targets
+   tags-ignore:
+     - release-*
+     - preview-*
+
+  pull_request:
+    branches: [ master ]
+    paths:
+      - src/HealthChecks.AzureKeyVault.Secrets/**
+      - test/HealthChecks.AzureKeyVault.Secrets.Tests/**
```
Comment (adamsitnik) [inline .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml:22]:
```suggestion
      - src/HealthChecks.Azure.KeyVault.Secrets/**
      - test/HealthChecks.Azure.KeyVault.Secrets.Tests/**
```

**Review by adamsitnik — COMMENTED** [review]
Overall it looks good, but we are missing a dot in the package name.

Thank you for providing the fix! With the next packages, I'll provide .yml file updates.

File: .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml (hunk 4 above)
Comment (unaizorrilla) [reply .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml]: Yeap sorry, i'm doing it before yours suggestion :-)

Comment (unaizorrilla) [reply .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml]: Hi @adamsitnik  

If it is ok, i will release a preview version to test it and the release if is working well from your side!

Comment (adamsitnik) [reply .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml]: @unaizorrilla Of course it's OK!

File: .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+          --no-restore
+          --no-build
+          --collect "XPlat Code Coverage"
+          --results-directory .coverage
+          --
+          DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=opencover
+      - name: Upload Coverage
+        uses: codecov/codecov-action@v3
+        with:
+          flags: AzureKeyVault
```
Comment (sungam3r) [inline .github/workflows/healthchecks_azurekeyvault_secrets_ci.yml:64]: The same flag as for existing AKV package?

**Pull request merged**